cantools/database/can/templates/DBC.h

```diff
@@ -234,4 +236,78 @@ class Signal {
   uint32_t spn_ {J1939_INVALID_SPN};
 };
 
+/** Partial template specialization for std::string PhysicalDataType */
+template<typename RawDataType>
+class Signal<RawDataType, std::string> {
+ public:
+  Signal(const uint8_t* buffer, const std::string& name)
```

Comment:
This gives you the buffer but not the length?

Reply:
Yeah, all code in this class is autogenerated, buffer is read-only. Can add length but would not be used.